### PR TITLE
CA-289145: close socket if error occurs when using lwt connect

### PR DESCRIPTION
**Pull request by krizex**
Signed-off-by: Yang Qian <yang.qian@citrix.com>

### CODE REVIEW

**Comment by mseri** [comment]
`-5%`?? Seriosuly @coveralls ? :confused:

**Comment by krizex** [comment]
Thanks @mseri , updated.

File: lwt/nbd_lwt_unix.ml [hunk 1]
```diff
@@ -97,7 +97,11 @@ let connect hostname port =
   Lwt_unix.gethostbyname hostname
   >>= fun host_info ->
   let server_address = host_info.Lwt_unix.h_addr_list.(0) in
-  Lwt_unix.connect socket (Lwt_unix.ADDR_INET (server_address, port))
+  Lwt.catch (fun () ->
+      Lwt_unix.connect socket (Lwt_unix.ADDR_INET (server_address, port))
+    ) (fun e ->
+      Lwt_unix.close socket >>= Lwt.fail e
```
Comment (mseri) [inline lwt/nbd_lwt_unix.ml]: `>>=` should be `>>= fun () ->`

**Comment by mseri** [comment]
The build error are due to incomplete upstream opam files

**Pull request merged**